container/factory.go

```diff
@@ -89,30 +86,27 @@ func HasFactories() bool {
 }
 
 // Create a new ContainerHandler for the specified container.
-func NewContainerHandler(name string, inHostNamespace bool) (ContainerHandler, bool, error) {
+func NewContainerHandler(name string, inHostNamespace bool) (ContainerHandler, error) {
 	factoriesLock.RLock()
 	defer factoriesLock.RUnlock()
 
 	// Create the ContainerHandler with the first factory that supports it.
 	for _, factory := range factories {
-		canHandle, canAccept, err := factory.CanHandleAndAccept(name)
+		handler, ignore, err := factory.NewContainerHandler(name, inHostNamespace)
```

Comment:
Consider: I like to avoid methods with more than 2 return parameters. What if we change the semantics to return if the factory returns an error? Then we can create an "IgnoreError", and simplify the logic here to just three cases: Return if `err != nil || handler != nil`, and continue if both are nil. You'd also need to update the factory methods to not return errors in the "expected" cases (e.g. in docker don't return an error when inspect fails) - which I think makes more sense anyway...